nibabel/tests/test_scripts.py

```diff
@@ -68,7 +68,15 @@ def check_nib_ls_example4d(opts=[], hdrs_str=""):
 def test_nib_ls():
     yield check_nib_ls_example4d
     yield check_nib_ls_example4d, \
-          ['-H', 'dim,bitpix'], " \[  4 128  96  24   2   1   1   1\] 16"
+        ['-H', 'dim,bitpix'], " \[  4 128  96  24   2   1   1   1\] 16"
+    yield check_nib_ls_example4d, ['-c'], "", " 2:3 3:2 4:1 5:1.*"
```

Comment:
On Wed, 06 Apr 2016, Matthew Brett wrote:

>    In nibabel/tests/test_scripts.py:

>  > @@ -68,7 +68,15 @@ def check_nib_ls_example4d(opts=[], hdrs_str=""):
>  >  def test_nib_ls():
>  >      yield check_nib_ls_example4d
>  >      yield check_nib_ls_example4d, \
>  > -          ['-H', 'dim,bitpix'], " \[  4 128  96  24   2   1   1   1\] 16"
>  > +        ['-H', 'dim,bitpix'], " \[  4 128  96  24   2   1   1   1\] 16"
>  > +    yield check_nib_ls_example4d, ['-c'], "", " 2:3 3:2 4:1 5:1.*"

>    Yes, the yields mean that you still run further tests if one fails, but
>    given we clean up our test errors quickly, that isn't worth that much. So,
>    I just do multiple assert_true(something) etc in the same test function.

I do regular asserts within test bodies as well not allowing to fall
through a failing one (so failures could cascade etc).

But in this case it is just a parametric test which invokes the same
body of the larger testing logic under various conditions.  A bit
different (although could be argued to be the same of cause just at a
different level).  I also like such tests to see the progress
which shows that a given test is validated against different
options/conditions.

Just my .1 cent, will later change to regular calls to the check_... but
in my experience such level of yields is imho only beneficial.  It was
always as easy to isolate or troubleshoot as if I would just be
calling those large test helpers.